File: wallet.py
class Wallet:
    def __init__(self):
        self.balance = 0
        self.transactions = []
    
    def get_balance(self):
        return self.balance
    
    def deposit(self, amount):
        if amount > 0:
            self.balance += amount
            self.transactions.append(f"Deposit: +{amount}")
            print("\nSuccessful deposit.")
        else: 
            print("\nPlease enter a valid amount!")
    
    def withdraw(self, amount):
        if self.balance >= amount:
            self.balance -= amount
            self.transactions.append(f"Withdraw: +{amount}")
            print("\nSuccessful withdrawal.")
        else:
            print("\nInsufficient funds!")

    def get_transactions(self):
        return self.transactions            

File: Wallet.cs
using System;
using System.Collections.Generic;

public class Wallet
{
    private decimal balance;
    private List<string> transactions;

    public Wallet()
    {
        balance = 0;
        transactions = new List<string>();
    
    }

    public decimal GetBalance()
    {
        return balance;
    }

    public void Deposit(decimal amount)
    {
        if (amount > 0)
        {
            balance += amount;
            transactions.Add($"Deposit: +{amount}");
            Console.WriteLine("\nSuccessful deposit.");
        }
        else 
        {
            Console.WriteLine("\nPlease enter a valid amount!");
        }
    }

    public void Withdraw(decimal amount)
    {
        if (balance >= amount)
        {
            balance -= amount;
            transactions.Add($"Withdrawal: -{amount}");
        }
        else 
        {
            Console.WriteLine("\nInsufficient funds!");
        }
    }

    public List<string> GetTransactions()
    {
        return transactions;
    }
}